### PR TITLE
[scanning] Implement new self-contained voucher entry flow (screens/scanning)

### PR DESCRIPTION
**Pull request by sauhardjain**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Redesigned the voucher entry flow to be self-contained. The task breakdown is given below:
- [x] Build `VoucherEntryStartScreen`
- [x] Implement success toasts
- [x] Add validation logic to `ScanningScreen`
- [x] Implement error alerts on `ScanningScreen`
- [x] Add exit button to voucher entry flow and create event handler (`handlePreventLeave`) to display an unsaved changes alert
- [x] Use the `fullScreenModal` presentation option to hide the bottom tab on screens within the voucher entry flow
- [x] Fix bug that shows the Transaction screens on the Add tab instead of the Invoices tab (after clicking Review Invoices)

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
https://www.loom.com/share/a17963b73b8e4b5a9768443ac6088d84

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Remove the hardcoded voucher value cap at $10.
- `VoucherCounter` has been temporarily removed (in favor of the exit button) but will be added back at a later point in time.

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
[React Navigation docs on screenOptions](https://reactnavigation.org/docs/native-stack-navigator/#options)
[React Navigation docs on type checking](https://reactnavigation.org/docs/typescript/#annotating-usenavigation)

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/screens/scanning/ConfirmValueScreen.tsx [hunk 1]
```diff
@@ -75,13 +68,22 @@ export default function ConfirmValueScreen({
...
+          onPress={() =>
+            handlePreventLeave({
+              hasUnsavedChanges,
+              navigation,
+              dispatch,
+            })
+          }
+        >
+          <Icon name="close" size={24} color={Colors.midBlack} />
+        </TouchableOpacity>
```
Comment (ronniebeggs) [inline src/screens/scanning/ConfirmValueScreen.tsx]: Should this button just return the user to the page they were just on? As of now, it exits the entire voucher entry flow. There may be instances where a user feels unsure about the serial number / barcode they just inputted.

File: src/utils/scanningUtils.ts [hunk 2]
```diff
@@ -28,3 +32,76 @@ export const deleteVoucher = (
...
+/** Displays a toast on successful voucher entry */
+export const showSuccessToast = () => {
+  Toast.show({
+    type: 'success',
+    position: 'top',
+    topOffset: 50,
+    text1: 'Voucher Added!',
+    visibilityTime: 2000,
+  });
+};
```
Comment (ronniebeggs) [inline src/utils/scanningUtils.ts:107]: Minor styling, but the text feels very small and may be hard to read on some devices.

File: src/screens/scanning/ReviewScreen.tsx [hunk 3]
```diff
@@ -116,9 +117,18 @@ export default function ReviewScreen({
...
+        index: 1,
+        routes: [
+          { name: 'VoucherEntryStartScreen' },
+          {
+            name: 'ConfirmationScreen',
+            params: { count: voucherMap.size },
+          },
+        ],
+      }),
+    );
```
Comment (ronniebeggs) [inline src/screens/scanning/ReviewScreen.tsx:133]: The edit amount modal is triggering "unhandled promise rejection" warnings, although it doesn't appear to be slowing down the app. Just noticed an odd message in the console.

**Review by ronniebeggs — COMMENTED** [review]
Overall, looks very good. A lot of cool new features for one PR. GOAT PL.

The only thing I felt throughout the app is that fonts felt very small/thin and may be difficult to view on some screens.

File: src/screens/scanning/ConfirmValueScreen.tsx (hunk 1 above)
Comment (sauhardjain) [reply src/screens/scanning/ConfirmValueScreen.tsx]: Great catch, this makes a lot of sense.

File: src/screens/scanning/ReviewScreen.tsx (hunk 3 above)
Comment (sauhardjain) [reply src/screens/scanning/ReviewScreen.tsx:133]: @selene-huang can you look into this?

File: src/utils/scanningUtils.ts (hunk 2 above)
Comment (sauhardjain) [reply src/utils/scanningUtils.ts:107]: Will implement this in a future release (TODO comment added)

File: src/screens/scanning/ReviewScreen.tsx (hunk 3 above)
Comment (selene-huang) [reply src/screens/scanning/ReviewScreen.tsx:133]: This seems to only be happening when you edit it to an invalid amount and the change gets rejected. We probably aren't catching something, but it doesn't seem to affect functionality too much.

**Comment by selene-huang** [comment]
LGTM!

File: src/screens/scanning/ReviewScreen.tsx (hunk 3 above)
Comment (selene-huang) [reply src/screens/scanning/ReviewScreen.tsx:133]: Should be handled now.
